crates/compiler/roar/src/storage.rs

```diff
@@ -0,0 +1,56 @@
+use roc_module::symbol::Symbol;
+use std::fmt::{Debug,Display};
+///The type used to identify registers
+pub(crate) type Id = u32;
+#[derive(Clone,Debug)]
+///A number of bytes an operation takes
+pub struct ByteSize(u32);
+///Equivalent to byte size, merely used as to specify the two
+pub(crate) type Offset = ByteSize;
+pub(crate) const WORD_SIZE : ByteSize = ByteSize(8);
+#[derive(Clone,Debug)]
+pub(crate) struct Register(pub Id);
+#[derive(Clone,Debug)]
+pub(crate) struct FloatRegister(pub Id);
+#[derive(Clone,Debug)]
+pub(crate) struct Global(Symbol);
+#[derive(Clone,Debug)]
+///The type of all literal values that fit inside a word
+pub enum LiteralValue {
+    Signed(i64),
+    Unsigned(u64),
+    Float(f64),
+    Char(char)
```

Comment:
I am not 100% sure your plan for literals, but a few notes:
1. we have integers upto 128 bit
2. float64 is layed out different than float32, which may be important in some case.
3. We have dec as the default fractional type which you can think of as an i128.
4. char doesn't exist in roc. something like `'a'` can be any integer type.